jsonschema_lexer/_lexer.py

```diff
@@ -24,109 +27,162 @@ class JSONSchemaLexer(JsonLexer):
-        "patternProperties",
-        "additionalProperties",
-        "dependentSchemas",
-        "propertyNames",
-        "prefixItems",
-        "contains",
-        "items",
-    ]
-    meta_data_keywords: ClassVar[list[str]] = [
-        "title",
-        "description",
-        "default",
-        "deprecated",
-        "examples",
-        "readOnly",
-        "writeOnly",
-    ]
-    validation_keywords: ClassVar[list[str]] = [
-        "type",
-        "enum",
-        "const",
-        "minLength",
-        "maxLength",
-        "pattern",
-        "maximum",
-        "exclusiveMinimum",
-        "multipleOf",
-        "exclusiveMaximum",
-        "minimum",
-        "dependentRequired",
-        "minProperties",
-        "maxProperties",
-        "required",
-        "minItems",
-        "maxItems",
-        "minContains",
-        "maxContains",
-        "uniqueItems",
-    ]
-    other_keywords: ClassVar[list[str]] = [
-        "format",
-        "unevaluatedItems",
-        "unevaluatedProperties",
-        "contentEncoding",
-        "contentMediaType",
-        "contentSchema",
-        "format_assertion",
+        '"object"',
+        '"integer"',
+        '"string"',
+        '"number"',
+        '"array"',
+        '"boolean"',
+        '"null"',
     ]
+    keywords: ClassVar[dict[str | None, list[str]]] = {}
+    identifier: ClassVar[dict[str | None, str]] = {}
+    default_dialect = None
+
+    def __init__(self, default_dialect: str | None = None):
+        super().__init__()  # type: ignore[reportUnknownMemberType]
+        self._populate_keywords_and_identifiers()
+        if default_dialect and default_dialect[0] != '"':
+            default_dialect = '"' + default_dialect
+
+        if default_dialect and default_dialect[-1] != '"':
+            default_dialect = default_dialect + '"'
+
+        self.default_dialect = default_dialect
+
+    def _populate_keywords_and_identifiers(self):
+        dialect_files = files("jsonschema_lexer") / "data" / "keywords"
+        if not dialect_files.is_dir():
+            dialect_files = Path(__file__).parent.parent / "data" / "keywords"
+        for dialect_file in dialect_files.iterdir():
+            with dialect_file.open() as file:
+                json_content = json.load(file)
+                dialect_name = self._make_string_double_quoted(
+                    json_content["dialect"],
+                )
+                self.keywords[dialect_name] = json_content["keywords"]
+                self.identifier[dialect_name] = (
+                    self._make_string_double_quoted(
+                        json_content["identifier"],
+                    )
+                )
+
+    def _make_string_double_quoted(self, string: str):
+        return '"' + string + '"'
+
+    def _find_rightmost_token_index(
+        self,
+        syntax_stack: list[tuple[int, str]],
+        token: str | None,
+    ):
+        for i in range(len(syntax_stack) - 1, -1, -1):
```

Comment:
```suggestion
return next((i for i, (_, t) in reversed(enumerate(syntax_stack)) if t == token), None)
```
or thereabouts looks like it should work here.

And similar below.